Cargo.toml

```diff
@@ -9,6 +9,5 @@ authors = [
     "cad97 <cad97@cad97.com>",
 ]
 
-# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
-
 [dependencies]
+serde = { version = "1.0", optional = true, features = ["derive"] }
```

Comment:
I wonder if implementing serde manually worth the improvement in compile times. It’s seems great If vocabulary crates don’t need to wait for syn to be build. OTOH, probability that all the crates use serde without derive feature seems small.

Hmm, actually, derived deserialize can violate start < end contract, so we better implement it manually anyway

Reply:
I hope you like manual deserialization code because that's a lot of it added to remove the derive dependency!

Normally, one would implement deserialize manually by deserializing a helper and then validating on that, but I do see the value in handwriting it to skip the derive dependency in a small leaf crate with only a couple types.